File: src/main/java/net/atos/zac/zoeken/model/zoekobject/ZoekObjectType.java
/*
 * SPDX-FileCopyrightText: 2021 - 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.zoeken.model.zoekobject;

public enum ZoekObjectType {
    TAAK(TaakZoekObject.class),
    ZAAK(ZaakZoekObject.class),
    DOCUMENT(DocumentZoekObject.class);

    private final Class<? extends ZoekObject> zoekObjectClass;

    ZoekObjectType(final Class<? extends ZoekObject> zoekObjectClass) {
        this.zoekObjectClass = zoekObjectClass;
    }

    public Class<? extends ZoekObject> getZoekObjectClass() {
        return zoekObjectClass;
    }

}

File: src/main/kotlin/nl/info/zac/search/model/zoekobject/ZoekObjectType.kt
/*
 * SPDX-FileCopyrightText: 2021 - 2022 Atos, 2024 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.zac.search.model.zoekobject

enum class ZoekObjectType(val zoekObjectClass: Class<out ZoekObject>) {
    // The order is important here.
    // For a full reindex we always want to reindex zaken first, before taken and documenten.
    ZAAK(ZaakZoekObject::class.java),
    TAAK(TaakZoekObject::class.java),
    DOCUMENT(DocumentZoekObject::class.java)
}
